packages/@storefront/flux-capacitor/test/unit/core/actions/validators.ts

```diff
@@ -652,4 +664,111 @@ suite('validators', ({ expect, spy, stub }) => {
       expect(validators.isNotFetching.func(forward, state)).to.be.true;
     });
   });
+
+  describe('hasValidLabels', () => {
+    it('should be valid if labels is not present', () => {
+      expect(validators.hasValidLabels.func(<any>{})).to.be.true;
+    });
+
+    it('should be valid if labels is an array of strings', () => {
+      expect(validators.hasValidLabels.func(<any>{ labels: ['a', 'b'] })).to.be.true;
+    });
+
+    it('should be valid if labels is an empty array', () => {
+      expect(validators.hasValidLabels.func(<any>{ labels: [] })).to.be.true;
+    });
+
+    it('should be invalid if labels is an array of anything other than strings', () => {
+      expect(validators.hasValidLabels.func(<any>{ labels: [{ foo: 'baz' }, 'bar'] })).to.be.false;
+    });
+  });
+
+  describe('hasValidOptions', () => {
+    it('should be valid if options is an array of objects with the required fields', () => {
+      const payload: any = {
+        options: [
+          { field: 'foo' },
+          { field: 'bar', descending: false },
+          { field: 'baz', descending: true },
+        ],
+      };
+
+      expect(validators.hasValidOptions.func(payload)).to.be.true;
+    });
+
+    [
+      'foo',
+      1,
+      true,
+      {},
+      () => null,
+      undefined,
+    ].forEach((val) => {
+      it(`should be invalid if options is of type ${typeof val}`, () => {
+        expect(validators.hasValidOptions.func(<any>{ options: val })).to.be.false;
+      });
+    });
+
+    it('should be invalid if options is an empty array', () => {
+      expect(validators.hasValidOptions.func(<any>{ options: [] })).to.be.false;
+    });
+
+    it('should be invalid if any of the options are missing the `field` key', () => {
+      const payload: any = {
+        options: [{ field: 'foo' }, { field: 'bar' }, { baz: 'quux' }],
+      };
+
+      expect(validators.hasValidOptions.func(<any>payload)).to.be.false;
+    });
+
+    it('should be invalid if any of the options contain a non-boolean value for `descending`', () => {
+      const payload: any = {
+        options: [{ field: 'foo', descending: true }, { field: 'bar', descending: 'baz' }],
+      };
+
+      expect(validators.hasValidOptions.func(<any>payload)).to.be.false;
+    });
+  });
+
+  describe('hasValidSelected', () => {
+    it('should be valid if selected is not present', () => {
+      expect(validators.hasValidSelected.func(<any>{})).to.be.true;
+    });
+
+    it('should be valid if selected is a number between 0 and options.length - 1', () => {
+      expect(validators.hasValidSelected.func(<any>{
+        options: ['a', 'b', 'c'],
+        selected: 0,
+      })).to.be.true;
+    });
+
+    it('should be invalid if selected is less than 0', () => {
+      expect(validators.hasValidSelected.func(<any>{
+        options: ['a', 'b', 'c'],
+        selected: -1,
+      })).to.be.false;
+    });
+
+    it('should be invalid if selected is greater than options.length - 1', () => {
```

Comment:
The description is precise, but maybe a bit too precise. Consider something like "past the end of the options array".

Reply:
Done, resolving feedback.